### PR TITLE
feat: implemented the StarSearch feedback component

### LINKED ISSUE #3296: feat: StarSearch feedback UI component
## Functionality
This feedback component is meant to give us information about whether users have found the replies of their prompts to be useful or not.

For the MVP we want to use it together with a PostHog capture event to understand user patterns. In the future we can make use of the widget to help fine-tune the model as well.

⚠️  **This issue refers to the component's UI, but the PostHog capture event is documented in this related issue #295** 

## Mockups

image

&nbsp;

image

#### **On hover**
image

```
import { GoThumbsdown } from "react-icons/go";
<GoThumbsdown />

import { GoThumbsup } from "react-icons/go";
<GoThumbsup />
```

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

This PR implements the StarSearch feedback feature.

## Related Tickets & Documents

Closes #3296 
Closes https://github.com/open-sauced/engineering/issues/295

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/833231/6f6d7262-d27b-49d5-85a3-718b26068a3e

## Steps to QA

1. You need to part of the cohort for the StarSearch feature flag to test this.
2. Go to `/star-search`
3. Ask a question.
4. Notice the new feedback buttons (thumbs up, thumbs down) at the end of the conversation.
5. Click either one.
6. A toast message thanks you for your feedback.
7. Open the devtools console in the browser and find the posthog events and expand them
8. Notice a new posthog event called `star_search_feedback`. (see video above to see it in action)

## Tier (staff will fill in)

- [ ] Tier 1
- [x] Tier 2
- [ ] Tier 3
- [ ] Tier 4

### CODE REVIEW

File: lib/hooks/useStarSearchFeedback.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+export const useStarSearchFeedback = () => {
+  const posthog = usePostHog();
+
+  const feedback = ({ feedback, promptContent, promptResponse }: StarSearchFeedbackAnalytic) => {
+    posthog.capture("star_search_feedback", {
+      feedback,
+      promptContent,
+      promptResponse,
```
Comment (jpmcb) [inline lib/hooks/useStarSearchFeedback.ts:16]: Perfect: is this data persisted in Posthog long term?

---

I've also wondered about just storing the queries / responses people are getting when they use it. Might be some valuable data in there regardless of if they're giving us feedback or not

Comment (nickytonline) [reply lib/hooks/useStarSearchFeedback.ts:16]: I'm not sure how long this data is persisted in Posthog.I can dig into that or maybe @isabensusan knows.

Comment (jpmcb) [reply lib/hooks/useStarSearchFeedback.ts:16]: We can also investigate a more full fledged solution via https://github.com/open-sauced/api/issues/751 if we want to start doing a chat "history" or "memory" of what was said before, we'll need this in the API anyways.

Comment (bdougie) [reply lib/hooks/useStarSearchFeedback.ts:16]: Story history in app and across workspaces is something came up in a demo today.

**Review by jpmcb — APPROVED** [review]

**Review by bdougie — APPROVED** [review]
Great work @nickytonline 

👍 this pr was helpful

**Pull request merged**